ApplicationService/connection_pool.py

```diff
@@ -0,0 +1,53 @@
+import os
+import psycopg2
+import threading
+from typing import Callable
+from ApplicationService.repositories.connectionInterface import *
+
+class postgresql_connection_pool(connection_pool):
+    def __init__(self, createConnection=None, max_connections=1, condition=None):
+        self.create_connection: Callable[[], connection] = createConnection or psycopg2_create_connection
+        self.allocated_connections: dict[int, tuple[connection, connection.cursor]] = dict()
+        self.max_connections: int = max_connections
+        self.free_connections: list[tuple[connection, connection.cursor]] = []
```

Comment:
Why the type is connection.cursor instead of cursor that you import?

Reply:
Thank you for remembering.